src/Xamarin.Android.Build.Tasks/Xamarin.Android.Common.targets

```diff
@@ -1947,6 +1947,19 @@ because xbuild doesn't support framework reference assemblies.
   <Touch Files="$(IntermediateOutputPath)android\bin\mono.android.jar" />
   
   <Touch Files="$(_AndroidStaticResourcesFlag)" AlwaysCreate="true" />
+
+  <ItemGroup>
+    <FileWrites Include="$(MonoAndroidIntermediate)android\src\mono\MonoRuntimeProvider.java"
+      Condition="Exists ('$(MonoAndroidIntermediate)android\src\mono\MonoRuntimeProvider.java')" />
+    <FileWrites Include="$(MonoAndroidIntermediate)android\src\mono\android\incrementaldeployment\MultiDexLoader.java"
+      Condition="Exists ('$(MonoAndroidIntermediate)android\src\mono\android\incrementaldeployment\MultiDexLoader.java')" />
+    <FileWrites Include="$(MonoAndroidIntermediate)android\src\mono\android\incrementaldeployment\IncrementalClassLoader.java"
+      Condition="Exists ('$(MonoAndroidIntermediate)android\src\mono\android\incrementaldeployment\IncrementalClassLoader.java')" />
+    <FileWrites Include="$(MonoAndroidIntermediate)android\src\mono\android\Seppuku.java"
+      Condition="Exists ('$(MonoAndroidIntermediate)android\src\mono\android\Seppuku.java')" />
+    <FileWrites Include="$(MonoAndroidIntermediate)android\src\mono\android\app\NotifyTimeZoneChanges.java"
+      Condition="Exists ('$(MonoAndroidIntermediate)android\src\mono\android\app\NotifyTimeZoneChanges.java')" />
+  </ItemGroup>
```

Comment:
Do we also need to add `$(MonoAndroidIntermediateAssetsDir)machine.config` and `mono.android.jar` to `FileWrites`?